tensorflow/contrib/layers/python/layers/layers_test.py

```diff
@@ -2553,6 +2641,18 @@ def testNHWCAndNCHWTrainingProduceSameOutput(self):
       self.assertAllClose(nhwc, nchw, atol=1e-4, rtol=1e-4)
 
 
+class BatchNormFloat16Test(BatchNormTest):
+  dtype = dtypes.float16
+
+
+class BatchNormFloat32Test(BatchNormTest):
+  dtype = dtypes.float32
+
+
+class BatchNormFloat64Test(BatchNormTest):
+  dtype = dtypes.float64
+
+
 class LayerNormTest(test.TestCase):
```

Comment:
This will make the tests substantially more expensive to run (3x the cost they have now).

I think it's best to triplicate a single representative test to cover all dtypes, and assume that because there's no dtype-specific code in the layers this is enough coverage.

What do you think?